### ISSUE TITLE
Feature Request: Use neovim-remote

### ISSUE THREAD

**Issue by erichlf** [arnaupv/nvim-devcontainer-cli#29]
Maybe it would be beneficial, especially with LSPs and such, to use [neovim-remote](https://neovim.io/doc/user/remote.html) to pass information to and from the devcontainer. The idea would be that with a neovim remote and client running data could be sent between the two to get information that wouldn't otherwise be available to the host neovim.

**Comment by arnaupv** [comment]
this is interesting as a possible way to work locally, but the idea of the devcontainer should be that's self-container and it works equally regardless of the host. That's why I like the concept of installing the dependencies inside the container. 

I like your approach as a feature it can work with a flag, but I do not think it should be the normal way to work. What do you think?

**Comment by erichlf** [comment]
I was just trying to come up with a way to work in a devcontainer that seems more like VSCode. The current metthod just opens a terminal and not even a new VI session. I just liked the idea of having my vim that I started in receiving information from the devcontainer as if it was working in the devcontainer.

For handling the install of nvim so as to bring it up in the devcontainer, I would recommend using features, and likely the `--additional-features` flag provided by devcontainer. You could combine that with the [apt package feature](https://github.com/rocker-org/devcontainer-features/tree/main/src/apt-packages) to install things in a debian based container.

**Comment by arnaupv** [comment]
Yeah, opening  a terminal instead of vim inside the devcontainer is a matter of taste. I agree with you in can be highly improved. Not sure how and I currently do not have time to work on that but PRs are very welcome 🚀 ! The [neovim-remote](https://neovim.io/doc/user/remote.html) sounds like the appropriate tool to do what you want to do.

For me, this is in the same category than this other issue #28 . All of them are valid ways to interact with the devcontainer, and the best approach is a matter of taste. Different options should be allowed and defined in the nvim plugin config.
